pkg/storage/mongo/mongo_test.go

```diff
@@ -97,6 +99,49 @@ func TestQueryKindNotFoundErrorCases(t *testing.T) {
 		require.Equal(t, errors.KindNotFound, errors.Kind(err))
 	}
 }
+
+func TestQueryKindUnexpectedErrorCases(t *testing.T) {
+	// Prepare error documents
+	// If there is any error in preparation phase, skip this test
+	ctx := context.Background()
+	mongoStorage := getStorage(t)
+	uri := os.Getenv("ATHENS_MONGO_STORAGE_URL")
+	client, err := mongo.Connect(ctx, options.Client().ApplyURI(uri))
+	if err != nil {
+		t.SkipNow()
+	}
+	defer client.Disconnect(ctx)
+	coll := client.Database("athens").Collection("modules")
+	errDocs := []struct {
+		Module  string      `bson:"module"`
+		Version string      `bson:"version"`
+		Mod     interface{} `bson:"mod"`
+		Info    interface{} `bson:"info"`
+	}{
+		{"model1", "v1.1.1", 12345678, "error document with integer mod"},
+		{"model2", "v2.0.0", true, "error document with boolean mod"},
+	}
+	// clear collection before insert
+	coll.Drop(ctx)
```

Comment:
can you please remove this drop? It can interfere with other parallel running tests

Reply:
Sure, I have removed the drop operation, and all the docs will be upserted instead.